src/preamble.js

```diff
@@ -2264,15 +2274,22 @@ function integrateWasmJS() {
     info['env'] = env;
     // handle a generated wasm instance, receiving its exports and
     // performing other necessary setup
-    function receiveInstance(instance) {
+    function receiveInstance(instance, module) {
       exports = instance.exports;
       if (exports.memory) mergeMemory(exports.memory);
       Module['asm'] = exports;
       Module["usingWasm"] = true;
-      removeRunDependency('wasm-instantiate');
+#if USE_PTHREADS
+      // Keep a reference to the compiled module so we can post it to the workers.
+      Module['wasmModule'] = module;
+#endif
+      if(!ENVIRONMENT_IS_PTHREAD) {
```

Comment:
`ENVIRONMENT_IS_PTHREAD` does not exist when not building with `-s USE_PTHREADS=1` mode, so this should be gated inside `if USE_PTHREADS`.

Reply:
This change (https://github.com/kripken/emscripten/pull/5710/files#diff-8f35bc569987bab9a3691d986d90e75fR72) makes it exist unconditionally (just like the rest of the `ENVIRONMENT_IS_FOO` checks).